### PR TITLE
feat: add bitfield command

### LINKED ISSUE #1150: Missing `BITFIELD` command
The `BITFIELD` command is listed in [the list of supported APIs](https://upstash.com/docs/redis/overall/rediscompatibility) (under Bitmap) but missing in the TypeScript SDK. It is however, available in the Python SDK.

In our project, we thought using the [`ioredis`](https://github.com/redis/ioredis) package could be a workaround but, as it's built on Node.js, it doesn't work in Cloudflare's serverless environments.

### PR DESCRIPTION
**Pull request by lewxdev**
resolves #1150

### CODE REVIEW

**Comment by lewxdev** [comment]
I tried to model the code after [the python implementation](https://github.com/upstash/redis-py/blob/b785df0c4ac9e3e15dc6eb0f46ce860f792312d2/upstash_redis/commands.py#L4233-L4296), but I can't get the tests to run. Could use an extra pair of eyes on it

File: pkg/commands/bitfield.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { type Requester } from "../http";
+import { Command, type CommandOptions } from "./command";
+
+type SubcommandArgs<Rest extends unknown[] = []> = [
```
Comment (ogzhanolguncu) [inline pkg/commands/bitfield.ts:4]: We generally prefix generics with`T`, so, `Rest` should be `TRest`. Let's make that change.

File: pkg/commands/bitfield.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ) {
+    super(["bitfield", ...cmd], opts);
+  }
+
+  private chain(...args: typeof this.command): this {
+    this.command.push(...args);
+    return this;
+  }
+
+  get(...args: SubcommandArgs) {
```
Comment (ogzhanolguncu) [inline pkg/commands/bitfield.ts:27]: Sorry I couldn't get ahead of my OCD 😄 `SubcommandArgs` -> `SubCommandArgs`

File: pkg/redis.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { createAutoPipelineProxy } from "../pkg/auto-pipeline";
 import {
   AppendCommand,
   BitCountCommand,
+  BitFieldCommand,
```
Comment (ogzhanolguncu) [inline pkg/redis.ts:5]: We should also add this to pipeline class. `pkg/pipeline.ts`

**Review by ogzhanolguncu — COMMENTED** [review]
Overall it looks great. There are just some minor convention issues.

**Comment by lewxdev** [comment]
In order to add the bitfield command to the pipeline, I had to split `BitFieldCommand` into variants for use with `redis` and `pipeline`. This way, `exec` can return the result or a chainable pipeline respectively

Also, one thing that can be addressed is being able to infer the length of the result `number[]` based on the commands called (similar to how pipeline works) but the current type may be fine for now

**Review by ogzhanolguncu — APPROVED** [review]
Looking awesome. Thanks.

**Review by CahidArda — APPROVED** [review]

**Pull request merged**